12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
+	* `transaction_size` is unreasonable.
+	* `customer_signature_algorithm` does not match any of the supported signing algorithms.
+	* `customer_signature` cannot be verified using `customer_public_key`.
+
+* SHOULD reject the appointment if:
+	* `dispute_delta` is too small.
+	* `fee_rate` is too low.
+
+If `accountability` is NOT being offered:
+
+* The receiving node MUST reject the appointment.
+
+Otherwise:
+
+* The receiving node SHOULD accept the appointment.
+
 #### Rationale
-The transaction `locator` can be deterministically computed by both the client and the server. Locators of wrong size are therefore invalid.
 
-`start_block` and `end_block` too close to the current chain tip may result in the tower missing the trigger and therefore should be avoided.
+The concept of too small for `dispute_delta` is subjective. The `dispute_delta` defines the time (in blocks) that the tower has to respond after a breach is seen. The smallest the value, the more the server risks to fail the appointment.
 
-Too far away is a subjective concept. Towers accepting jobs that will start in the far future or that may last a really long time risk having to store data for long periods of time.
+`transaction_size` and `transaction_fee` help the WatchTower to decide on the likelihood of an appointment being fulfilled. Appointments with `fee_rate` too low may be rejected by the WatchTower. While a customer can always fake this values, it should break ToS between the client and the server and, therefore, release the WatchTower of any liability.
 
-The concept of too small for `dispute_delta` is also subjective. The `dispute_delta` defines how many blocks the server will have to respond with the `justice_transaction`. The smallest the value, the more the server risks to fail the appointment.
+If `accountability` is not being offered, it makes not much sense accepting appointments that request it. If the tower accepts an appointment requesting `accountability`, it should be enforced or refunded. Generally, this is trying to allow a reputationally accountable watching service. The signed job from the customer provides an explicit acknowledgement of the transaction details that is important for the WatchTower to decide whether it can accept them. If the decrypted justice transaction does not satisfy the signed job (e.g. fee too low), then the WatchTower is not obliged to fulfil the appointment. 
 
-`encrypted_blob` should have been encrypted using `cipher`. Block ciphers have a size multiple of the block length, which depends on the key size. Therefore some incorrect `encrypted_blob` can be spotted checking the `transaction_size`. Moreover, `encrypted_blob` have to be at least as big as:
+#### `tower_evidence`
 
-`ceil(minimum_viable_transaction_size / cipher_block_size)`
+The format for the `tower_evidence` is defined as follows:
 
-And at most as big as:
+1. type: ? (`tower_evidence`)
+2. data:  
+	* [`u16 `:`receipt_length`]
+	* [`receipt_length `: `receipt`]
```

Comment:
Should this be `receipt_length*byte`?

Reply:
Also perhaps uniformize the naming, some fields have _len and some _length.

Reply:
Agreed